backend/messaging-api/messaging-api.yaml

```diff
@@ -0,0 +1,190 @@
+openapi: 3.0.1
+info:
+  title: Feature Messaging API
+  description: This is an OpenAPI document for featurehub messaging data. It is used to publish feature updates to messaging listeners.
+  version: "1.1.1"
+paths:
+  /v2/fake:
+    get:
+      operationId: fakeApi
+      tags:
+        - FeatureMessagingService
+      description: "this is just a holder, it is not generated"
+      responses:
+        200:
+          description: "yay"
+components:
+  schemas:
+    FeatureMessagingUpdate:
+      x-cloudevent-type: messaging-feature-v1
+      x-cloudevent-subject: io.featurehub.events.messaging
+      x-publish-include: messaging
+      x-package: io.featurehub.messaging.model
+      description: "Changes made to a specific feature and who changed it"
+      type: object
+      required:
+        - environmentId
+        - applicationId
+        - whenUpdated
+        - whoUpdated
+        - featureKey
+        - whatUpdated
+      properties:
+        targetDestination:
+          description: "If there is a specific target for the data, everything else _should_ ignore it"
+          type: string
+        environmentId:
+          x-basename: eId
+          type: string
+          format: uuid
+        portfolioId:
+          x-basename: pId
+          type: string
+          format: uuid
+        organizationId:
+          x-basename: oId
+          type: string
+          format: uuid
+        applicationId:
+          x-basename: aId
+          type: string
+          format: uuid
+        whenUpdated:
+          type: string
+          format: date-time
+        whoUpdated:
+          type: string
+          description: Name of the user who made the change to the feature
+        featureKey:
+          description: "Feature key that got updated"
+          type: string
+        featureValueUpdated:
+          description: "Feature value changes"
+          $ref: '#/components/schemas/MessagingFeatureValueUpdate'
+        lockUpdated:
+          description: "Lock changes"
+          $ref: '#/components/schemas/MessagingLockUpdate'
+        retiredUpdated:
+          description: "Retired changes"
+          $ref: '#/components/schemas/MessagingRetiredUpdate'
+        strategiesUpdated:
+          description: "Strategies changes"
+          type: array
+          items:
+            $ref: '#/components/schemas/MessagingStrategyUpdate'
+        strategiesReordered:
+          description: "Strategies reorder changes"
+          $ref: '#/components/schemas/MessagingStrategiesReorder'
+        additionalInfo:
+          description: "Any additional info, webhook headers etc"
+          type: array
+          items:
+            type: string
+
+    MessagingFeatureValueUpdate:
+      x-package: io.featurehub.messaging.model
+      type: object
+      description: "Feature value changes, if any"
+      required:
+        - valueType
```

Comment:
the value type is part of the feature itself, for example, when a strategy updates, it may alter the value when it matches. So the valueType is important in the "value" associated with the default as well as individual strategies.